### PR TITLE
Feat: Code Component

### PR DESCRIPTION
**Pull request by Rain-Hendrix**
**Summary**

This PR creates a code component for Compound

This PR fixes/implements the following **bugs/features**

- Creates a Code component

When a user wants to use a code bracket, or single line of code. They can use this Atom to show their code styled with the correct Markup

**Documentation Update (required)**

If this pull request requires a change to Emulsify documentation, those changes, updates, and/or new information must accompany this pull request.

**How to review this PR**
- [x] Go here -> http://localhost:6006/?path=/story/atoms-text--code-example
- [ ] Use your code editor to go here -> `components/01-atoms/text/code` to view the files to see if the conventions that I used was correct

### CODE REVIEW

**Review by callinmullaney — CHANGES_REQUESTED** [review]
@Gregg-Hendrix HTML for this PR looks pretty good. However, I think you're missing changes to the `text.stories.js` file as I'm not actually able to view this component in Storybook.

Also, instead of using the BEM function directly on an element it's best used within a `pre__attributes` and `code__attributes` array. Then use the `add_attributes()` function directly on the element. 

Setting up the component this way allows for other (non-class attributes) to be added to markup whenever they are include/embed via another component. See https://github.com/emulsify-ds/compound/pull/39/files#diff-61519f74d1123aa5bd9ca2b2c84c23d9abe0ad31ed001d5cd7b451b74b3cdb64R15 as an example.

File: components/01-atoms/text/text/07-code.twig [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ */
+#}
+
+{% set code__base_class = code__base_class|default('code') %}
+{% set code__attributes = code__attributes|default({}) %}
+{% set code__attributes = code__attributes|merge({
+  'class': bem(code__base_class, code__modifiers, code__blockname),
+}) %}
+
+<pre {{ bem(code__base_class, code__modifiers, code__blockname) }}>
```
Comment (callinmullaney) [inline components/01-atoms/text/text/07-code.twig:18]: This markup should also use the add_attributes function similar to how you setup the `code__attributes`.

```
{{ add_attributes(pre__attributes) }}
```

Comment (callinmullaney) [reply components/01-atoms/text/text/07-code.twig:18]: In addition to a `pre__attributes` we should also define a separate `pre__base_class`

**Review by callinmullaney — CHANGES_REQUESTED** [review]
@Gregg-Hendrix This component is coming along. See my code comments below. I also think there should be a small bit of styles applied here to help differentiate the code display. Something like 
```
.pre {
  padding: $space;
}

.code {
  background-color: clr(background-section);
}
```

File: components/01-atoms/text/text/07-code.twig (hunk 1 above)
Comment (Gregg-Hendrix) [reply components/01-atoms/text/text/07-code.twig]: Thank you for this feedback Callin! I like how the background color only highlights the code now! So it does stand out. Before the whole component looked like "code". Now currently only the lines of "code" look like code! It was also helpful to see how attributes are defined, and then used on the component itself :).

**Comment by Gregg-Hendrix** [comment]
> @Gregg-Hendrix approving this PR so we can review/approve as a group during our Frontend L10. Can you update the styles one more time?
> 
> 1. Remove the styles added to `.code`
> 2. Add `margin: 0;` and `background-color: var(--c-background-section);` to `.pre`.
> 
> If you make these changes you should see it like: Screenshot from 2022-08-21 23-07-10

@callinmullaney , thank you for this feedback! I have now made these changes, and it matches the photo. I will add this to the L10 IDS :)

**Pull request merged**